contracts/dk/DuelistKingStaking.sol

```diff
@@ -0,0 +1,253 @@
+    ERC20 currentToken = ERC20(currentCampaign.tokenAddress);
+
+    require(currentUserStakingSlot.stakingAmountOfToken > 0, 'DKStaking: No token to be unstaked');
+
+    /**
+     * User unstake before endDate
+     * will be paid for penalty fee: 2% DKT and 50% reward box
+     */
+    currentUserStakingSlot.stakedAmountOfBoxes = estimateUserReward(currentCampaign, currentUserStakingSlot);
+
+    if (block.timestamp < currentCampaign.endDate) {
+      uint256 penaltyAmount = (currentUserStakingSlot.stakingAmountOfToken * 2) / 100;
+      _totalPenalty[currentCampaign.tokenAddress] += penaltyAmount;
+      currentToken.safeTransfer(msg.sender, currentUserStakingSlot.stakingAmountOfToken - penaltyAmount);
+
+      emit ClaimRewardBoxes(
+        msg.sender,
+        currentUserStakingSlot.stakedAmountOfBoxes / (2 * decimals),
+        currentCampaign.rewardPhaseBoxId
+      );
+      emit Unstaking(msg.sender, currentUserStakingSlot.stakingAmountOfToken - penaltyAmount, uint64(block.timestamp));
+    } else {
+      currentToken.safeTransfer(msg.sender, currentUserStakingSlot.stakingAmountOfToken);
+      emit ClaimRewardBoxes(
+        msg.sender,
+        currentUserStakingSlot.stakedAmountOfBoxes / decimals,
+        currentCampaign.rewardPhaseBoxId
+      );
+      emit Unstaking(msg.sender, currentUserStakingSlot.stakingAmountOfToken, uint64(block.timestamp));
+    }
+
+    // remove user staking amount from the pool
+    currentCampaign.stakedAmountOfToken -= currentUserStakingSlot.stakingAmountOfToken;
+
+    currentUserStakingSlot.stakingAmountOfToken = 0;
+    currentUserStakingSlot.stakedAmountOfBoxes = 0;
+
+    // Update campaign and user staking slot
+    _campaignStorage[campaignId] = currentCampaign;
+    _userStakingSlot[campaignId][msg.sender] = currentUserStakingSlot;
+    return true;
+  }
+
+  /*******************************************************
+   * Same domain section
+   *******************************************************/
+
+  // Create a new staking campaign
+  function createNewStakingCampaign(StakingCampaign memory newCampaign)
+    external
+    onlyAllowSameDomain('StakingOperator')
+  {
+    require(
+      newCampaign.startDate > block.timestamp && newCampaign.endDate > newCampaign.startDate,
+      'DKStaking: Invalid timeline format'
+    );
+    uint64 duration = (newCampaign.endDate - newCampaign.startDate) / (1 days);
+    require(duration > 1, 'DKStaking: Duration must be greater than 1');
+
+    require(newCampaign.rewardPhaseBoxId >= 1, 'DKStaking: Invalid phase id');
+    require(newCampaign.tokenAddress.isContract(), 'DKStaking: Token address is not a smart contract');
+
+    newCampaign.returnRate = uint64(
+      (newCampaign.maxNumberOfBoxes * decimals) / (newCampaign.maxAmountOfToken * duration)
+    );
+    _campaignStorage[_totalCampaign] = newCampaign;
+
+    emit NewCampaign(newCampaign.startDate, newCampaign.tokenAddress, _totalCampaign);
+    _totalCampaign += 1;
```

Comment:
You might need to `return true;` or return campaign id to allow external call get the result or checking for error.